src/luvi.c

```diff
@@ -18,10 +18,28 @@
 #include "./luvi.h"
 
 LUALIB_API int luaopen_luvi(lua_State *L) {
+  int i = 1;
   lua_newtable(L);
 #ifdef LUVI_VERSION
   lua_pushstring(L, ""LUVI_VERSION"");
   lua_setfield(L, -2, "version");
 #endif
+  lua_newtable(L);
+#ifdef WITH_OPENSSL
+  lua_pushnumber(L, i++);
+  lua_pushstring(L, "ssl");
+  lua_settable(L, -3);
+#endif
+#ifdef WITH_ZLIB
+  lua_pushnumber(L, i++);
+  lua_pushstring(L, "zlib");
+  lua_settable(L, -3);
```

Comment:
Not that this code is performance sensitive, but in this case it's better to use http://www.lua.org/manual/5.1/manual.html#lua_rawseti

Then you don't need to push wrapped numbers on the stack and skips a lot of steps internally.